src/tools/chip-cert/CertUtils.cpp

```diff
@@ -216,38 +216,53 @@ void ToolChipDN::PrintDN(FILE * file, const char * name) const
 
 namespace {
 
+template <size_t N>
+bool HasRawPrefix(const uint8_t * buffer, size_t len, const uint8_t (&prefix)[N])
+{
+    return len >= N && memcmp(buffer, prefix, N) == 0;
+}
+
+bool HasStringPrefix(const uint8_t * buffer, size_t len, const char * prefix)
+{
+    size_t prefixLen = strlen(prefix);
```

Comment:
Please strnlen to be max of `len`, since otherwise it's not relevant.

Reply:
The next line checks for `len >= prefixLen` since we want the entire prefix to be present, so we don't want to pre-truncate the prefix to be a max of `len`.

Reply:
My point is that a "free" `strlen` without length cap is frowned-upon in this codebase. This is probably fine here, though.